src/data/markdown/translated-guides/en/03 Results output/200 Real-time/00 Prometheus.md

```diff
@@ -1,63 +1,121 @@
 ---
-title: 'Prometheus'
-excerpt: 'The Prometheus integration allows to send test results to any Prometheus Remote Write endpoint.'
+title: 'Prometheus remote write'
+excerpt: 'Use the Prometheus remote write integration to send test results to any Prometheus remote write endpoint.'
 ---
 
-k6 supports sending test result metrics to a Prometheus Remote Write endpoint via the [`xk6-output-prometheus-remote` extension](https://github.com/grafana/xk6-output-prometheus-remote).  One option with this support is storing the metrics in Prometheus; others can be found [here](https://prometheus.io/docs/operating/integrations/).
+<ExperimentalBlockquote />
 
-## Instructions
+Prometheus remote write is a protocol with a defined [specification](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit).
+It has multiple implementations.
+For example, one option is to store the metrics in [Prometheus](https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver).
+You can find other implementations in the Prometheus' [Integrations](https://prometheus.io/docs/operating/integrations) guide.
 
-<InstallationInstructions extensionUrl="github.com/grafana/xk6-output-prometheus-remote"/>
+With the Prometheus remote write output, k6 can send test-result metrics to a Prometheus remote write endpoint.
+The output during the `k6 run` execution gets all the generated data points for the [built-in  k6 metrics](/using-k6/metrics/).
+It then generates the equivalent Prometheus remote write time series.
 
+## Metrics mapping
 
-Then run the test with the new binary as follows:
+All the k6 metric types are converted into an equivalent Prometheus remote write type:
 
+| k6 | Prometheus | Name label |
+|----|------------| ---------- |
+| Counter | Counter | `k6_*_total`
+| Gauge | Gauge | `k6_*_<unit-suffix>` |
+| Rate | Gauge | `k6_*_rate` |
+| Trend | Gauges / Native Histogram | `k6_*_<unit-suffix>` |
+
+### Trend
+
+The obvious conversion with a classic Prometheus Histogram is not convenient because k6 can't determine the fixed buckets in advance.
```

Comment:
>The first solution is an "hack" and technically is Trend-to-Counters-and-Gauges

Also, we aren't doing Trend to "classic" Histogram because it would require the user to specify the static buckets and it would create a bad UX.

Reply:
```
## Trend metric conversion

By default, k6 trend metrics convert to primitive counters and gauges. To convert trend metrics to high-fidelity histograms, you can use Prometheus Native histogram instead.

Because k6 can't easily determine fixed buckets in advance, k6 metrics can't export to the original Prometheus histograms.
So, the output maps a trend metric into primitive counter and gauges.
Each value represents a math function (count, sum, min, max, avg, med, p(x)).
This mapping has following drawbacks:
- It is impossible to aggregate some gauge values (especially percentiles).
- It uses a memory-expensive k6 data structure.

To resolve these limitations, you can map a Trend as a [Prometheus Native Histogram].(https://prometheus.io/docs/concepts/metric_types/#histogram).
To do this, you can use the `K6_PROMETHEUS_RW_TREND_AS_NATIVE_HISTOGRAM=true` environment variable (or one of the other ways).
The output then converts all the trend types into a dedicated Native Histogram.

Mapping trends as native histograms gives better you more efficient storage and preciser queries.
The drawback is that **the feature is experimental, released in Prometheus v2.40.0**.

<Blockquote mod="note" title="">

Native Histogram is an experimental feature, so it has to be enabled ([--enable-feature=native-histograms](https://prometheus.io/docs/prometheus/latest/feature_flags/#native-histograms)). Note that other remote write implementations don't support it yet.

</Blockquote>

```

What do you think about that? Writing that made it clearer to me, but I don't know about other readers.